integrators/ordinary_differential_equations.hpp

```diff
@@ -37,9 +42,13 @@ struct SpecialSecondOrderDifferentialEquation {
     std::vector<Displacement> position_error;
     std::vector<Velocity> velocity_error;
   };
+  // A functor that computes f(q, t) and stores it in |*accelerations|.
+  // |*accelerations| must have the same size as |positions|.
```

Comment:
On return, |*accelerations|...

Also, do we ensure anything about the input value of accelerations, eg it's empty?  Or it has the right number of 0 accelerations?